File: Design Patterns in Java/Chain of Responsibility Pattern/src/dev/chamalwr/handler/DomainHandler.java
package dev.chamalwr.handler;

import dev.chamalwr.model.Email;

/**This handler responsible of checking
 * the email is coming from the the internal mail system
 * of the company
 **/
public class DomainHandler extends EmailHandler {

    private String companyDomain =  "luxWatches.com";

    @Override
    public void checkMail(Email email) {
        if(email.getSenderDomain().equals(companyDomain)){
            System.out.println("This is an Internal Email from Lux Watches");
            emailHandlerSuccessor.checkMail(email);
        }else{
            System.err.println("This email is from another organization!");
        }
    }
}

File: Design Patterns in JavaScript/Chain of Responsibility Pattern/handler/DomainHandler.js
const EmailHandler = require("./EmailHandler");

module.exports = class DomainHandler extends EmailHandler{
    companyDomain = "luxWatches.com";

    checkMail(email){
        if (this.companyDomain === email.senderDomain){
            console.log("This is an Safe internal email")
            if (this.successorHandler != null){
                this.successorHandler.checkMail(email)
            }
        }else {
            console.log("This is an External Email!")
        }
    }
}
